### PR TITLE
feat: Picture optimization parameters added to enrich.yml

### PR DESCRIPTION
**Pull request by almtrimech**
| Q           | A                                               |
|-------------|-------------------------------------------------|
| Ticket      | https://redmine.aramisauto.com/issues/50034      |
| Change type | Build |

### CODE REVIEW

File: marketplace/picture-optimizer.schema.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                "cropping-margin": {
+                    "type": "boolean"
+                },
+                "cropping-margin-ratio": {
+                    "type": "object",
+                    "description": "List of cropping margin auto",
+                    "additionalProperties": false,
+                    "properties": {
+                        "top" : {
+                            "type": "number"
```
Comment (linepogl-aa) [inline marketplace/picture-optimizer.schema.json]: What's the unit for this? Pixels? Percent?

File: marketplace/picture-optimizer.schema.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+        "operations": {
+            "type": "object",
+            "additionalProperties": false,
+            "description": "List of operations (filters) to apply on picture",
+            "properties": {
+                "cropping": {
+                    "type": "boolean"
```
Comment (linepogl-aa) [inline marketplace/picture-optimizer.schema.json]: Shouldn't this be an object `{ active: true }` like the others ?

File: marketplace/picture-optimizer.schema.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                },
+                "cropping-margin": {
+                    "type": "boolean"
```
Comment (linepogl-aa) [inline marketplace/picture-optimizer.schema.json]: same

File: marketplace/picture-optimizer.schema.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                "keywordsBlurring": {
+                    "type": "object",
+                    "description": "List of keywords for blurring",
+                    "additionalProperties": false,
+                    "properties": {
+                        "active": {
+                            "type": "boolean"
+                        },
+                        "keywords" : {
+                            "type": "object",
```
Comment (linepogl-aa) [inline marketplace/picture-optimizer.schema.json]: shouldn't this be an array of strings?

File: marketplace/picture-optimizer.schema.json [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                            "minProperties": 1
+                        }
+                    },
+                    "if": {"properties": {"active" : true}},
```
Comment (linepogl-aa) [inline marketplace/picture-optimizer.schema.json]: 👍

File: marketplace/picture-optimizer.schema.json [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                    "maxProperties": 2
+                },
+                "plate": {
+                    "type": "boolean"
+                },
+                "backgroundRemoval": {
+                    "type": "boolean"
+                }
+            },
+            "minProperties": 1,
```
Comment (linepogl-aa) [inline marketplace/picture-optimizer.schema.json]: Is it possible to have no operation at all? Can we disable image processing altogether for a partner?

File: marketplace/enrich.schema.json [hunk 7]
```diff
@@ -27,6 +27,79 @@
...
                     "default": false
+                },
+                "pictureOptimizer": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "minProperties": 1,
+                    "maxProperties": 2,
+                    "description": "Configuration to apply picture optimizer",
+                    "properties": {
+                        "active": false,
```
Comment (linepogl-aa) [inline marketplace/enrich.schema.json]: `{ type: boolean, default: ... }`

File: marketplace/picture-optimizer.schema.json (hunk 6 above)
Comment (almtrimech) [reply marketplace/picture-optimizer.schema.json]: yes, we can disable processing for a partner, so when it's enabled, we should have at least one operation

File: marketplace/picture-optimizer.schema.json (hunk 1 above)
Comment (almtrimech) [reply marketplace/picture-optimizer.schema.json]: percent 0 to 1

Comment (linepogl-aa) [reply marketplace/picture-optimizer.schema.json]: Add `minimum` and `maximum`

File: marketplace/enrich.schema.json [hunk 8]
```diff
@@ -9,24 +9,193 @@
             "additionalProperties": false,
             "properties": {
                 "weight": {
-                    "type": "boolean",
-                    "default": false
+                    "type": "object",
+                    "additionalProperties": false,
+                    "description": "Enabled/disable weight enricher",
```
Comment (gmacaire-aa) [inline marketplace/enrich.schema.json]: IMO, Enricher parameters description should provide general information about the operation that is done on the data, not a message indicating that it is possible to activate or desactivate the operation. ;)

for this specific operation I'd say something like : _"Try to fill in vehicle weight if not provided by the seller"_

Same comment for other params. :)

File: marketplace/enrich.schema.json [hunk 9]
```diff
@@ -9,24 +9,193 @@
...
+                            "default": false
+                        }
+                    },
+                    "required": ["enabled"]
                 },
                 "category": {
-                    "type": "boolean",
-                    "default": false
+                    "type": "object",
+                    "additionalProperties": false,
```
Comment (gmacaire-aa) [inline marketplace/enrich.schema.json:25]: I didn't know the `additionalProperties` parameter. Thanks for helping me improve my knowledges. :-p

File: marketplace/enrich.schema.json [hunk 10]
```diff
@@ -9,24 +9,193 @@
...
+                },
+                "pictureOptimizer": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "description": "Enabled/disable picture optimizer enricher",
+                    "properties": {
+                        "enabled": {
+                            "type": "boolean",
+                            "default": false,
+                            "description": "By default, picture optimization is disabled for all partners"
```
Comment (gmacaire-aa) [inline marketplace/enrich.schema.json:79]: I wouldn't speak about "all partners" here. This parameter here only applies to one seller (partner).

File: marketplace/enrich.schema.json [hunk 11]
```diff
@@ -9,24 +9,193 @@
...
+                        "crop": {
+                            "type": "object",
+                            "additionalProperties": false,
+                            "properties": {
+                                "enabled": {
+                                    "type": "boolean",
+                                    "default": false
+                                }
+                            },
+                            "required": [
```
Comment (gmacaire-aa) [inline marketplace/enrich.schema.json]: cometic : short arrays (length to be defined) should maybe be written on a single row. What do you think ?

File: marketplace/enrich.schema.json [hunk 12]
```diff
@@ -9,24 +9,193 @@
...
+                    "description": "Enabled/disable picture optimizer enricher",
+                    "properties": {
+                        "enabled": {
+                            "type": "boolean",
+                            "default": false,
+                            "description": "By default, picture optimization is disabled for all partners"
+                        },
+                        "crop": {
+                            "type": "object",
+                            "additionalProperties": false,
```
Comment (gmacaire-aa) [inline marketplace/enrich.schema.json:83]: Any description here ? :)

If I don't now anything about this parameter, what should I expect if I activate it ?

**Review by gmacaire-aa — CHANGES_REQUESTED** [review]
Seems aligned with our workshops and discussions, thank you. :)

Could you please just ensure to add a description to each enrichment operation so that the user can have a little bit of documentation in its text editor at the time of writing config ?

**Review by gmacaire-aa — APPROVED** [review]

**Pull request merged**